samwise.js

```diff
@@ -15,57 +15,61 @@
     var timeoutHandle = null;
```

Comment:
SCRIPT-This block of code could benefit from some comments explaining the logic behind it.